vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/LinkUrlSearch.java

```diff
@@ -77,6 +81,62 @@ private static String buildExpectedLinkUrl(String linkUrl)
                 ? linkUrl + "/" : linkUrl;
     }
 
+    private static String getCurrentHref(String href, String url, String currentUrl)
+    {
+        if (href == null || url == null)
+        {
+            return null;
+        }
+        boolean startsWithSlash = url.charAt(0) == '/';
+        Pattern pattern = Pattern.compile(buildCurrentDomainPattern(startsWithSlash, currentUrl));
+        if (startsWithSlash)
+        {
+            Matcher m = pattern.matcher(href);
+            if (m.find())
+            {
+                return href.substring(m.group(1).length());
+            }
+        }
+        else if (!pattern.matcher(url).find() && areSchemeAndAuthorityEqual(currentUrl, href))
+        {
+            if (url.charAt(0) == SHARP_SYMBOL)
+            {
+                return href.substring(href.indexOf(SHARP_SYMBOL));
+            }
+            Matcher m = pattern.matcher(currentUrl);
+            if (m.find())
```

Comment:
not all branches of this condition is covered with unit tests, see previous screenshot of the coverage: yellow mean partial coverage